### DISCUSSION TITLE
How to switch between layout?

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7068] [Q&A] [answered]
In `$:/snippets/LayoutSwitcher`, seems we ca use `<$link>` widget to switch between layouts?

```php
<$link to={{!!title}}>
''<$transclude field="name"/>''&#32;-&#32;<$transclude field="description"/>
</$link>
```

But after the switch, the URL remains the same, and can't use "Back" to go back. How to implement a button to switch to default layout and open a tiddler, then allow user quickly back to previous layout?

**Comment by saqimtiaz** [comment, marked as answer]
> seems we ca use `<$link>` widget to switch between layouts?

That works because there is a surrounding linkcatcher widget.

To change layouts, change the value of the tiddler `$:/layout`

`<$action-setfield tiddler="$:/layout" text="myLayoutPageTemplate"/>`

**Reply by linonetwo** [reply]
Thanks, I omit the `<$linkcatcher to="$:/layout">` above.

So maybe I can quit my layout and open a tiddler in default layout using

```php
<$button tooltip="Open In StoryView">
  {{$:/core/images/home-button}}
  <$action-setfield tiddler="$:/layout" text="$:/core/ui/PageTemplate"/>
  <$action-navigate $to={{$:/state/Whiteboard/PageLayout/tiddler}}/>
</$button>
```

Currently it is not working, I'm still trying to figure out...

**Reply by saqimtiaz** [reply]
Apologies, there is a typo in my previous post, the tiddler attribute is actually $tiddler.

`<$action-setfield $tiddler="$:/layout" text="myLayoutPageTemplate"/>`

**Reply by linonetwo** [reply]
Got it, it's `<$action-setfield $tiddler=`, with `$tiddler=`, not `tiddler=`, problem solved! Thank you again Saq!

**Comment by linonetwo** [comment]
About how to go back, I add a button in the tiddler toolbar, so we can go back and forth:

https://user-images.githubusercontent.com/3746270/204126586-932f3676-9131-4859-a901-5de8fdcbfddd.mov

**Reply by saqimtiaz** [reply]
Very nice. What library are you using for the whiteboard?

**Reply by linonetwo** [reply]
https://github.com/tldraw , same as the https://github.com/toeverything/AFFiNE
